bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/actions/MatterOTBRActions.java

```diff
@@ -0,0 +1,304 @@
+        NodeHandler handler = this.handler;
+        if (handler != null) {
+            ThreadBorderRouterManagementConverter converter = handler
+                    .findConverterByType(ThreadBorderRouterManagementConverter.class);
+            if (converter != null) {
+                try {
+                    ThreadDataset tds = null;
+                    if (dataset.trim().startsWith("{")) {
+                        tds = ThreadDataset.fromJson(dataset);
+                        if (tds == null) {
+                            return ResourceHelper
+                                    .getResourceString(MatterBindingConstants.THING_ACTION_RESULT_INVALID_JSON);
+                        }
+                    } else {
+                        tds = ThreadDataset.fromHex(dataset);
+                    }
+                    converter.updateThreadConfiguration(tds.toHex());
+                    return ResourceHelper.getResourceString(MatterBindingConstants.THING_ACTION_RESULT_SUCCESS);
+                } catch (Exception e) {
+                    logger.debug("Error setting  dataset", e);
+                    return "error: " + e.getMessage();
+                }
+            } else {
+                return ResourceHelper.getResourceString(MatterBindingConstants.THING_ACTION_RESULT_NO_CONVERTER);
+            }
+        } else {
+            return ResourceHelper.getResourceString(MatterBindingConstants.THING_ACTION_RESULT_NO_HANDLER);
+        }
+    }
+
+    @RuleAction(label = MatterBindingConstants.THING_ACTION_LABEL_OTBR_LOAD_ACTIVE_DATASET, description = MatterBindingConstants.THING_ACTION_DESC_OTBR_LOAD_ACTIVE_DATASET)
+    public @ActionOutputs({
+            @ActionOutput(name = "result", label = MatterBindingConstants.THING_ACTION_LABEL_OTBR_LOAD_ACTIVE_DATASET_RESULT, type = "java.lang.String"),
+            @ActionOutput(name = "dataset", label = MatterBindingConstants.THING_ACTION_LABEL_OTBR_LOAD_ACTIVE_DATASET_DATASET, type = "java.lang.String") }) Map<String, Object> loadActiveOperationalDataset() {
+        NodeHandler handler = this.handler;
+        if (handler != null) {
+            ThreadBorderRouterManagementConverter converter = handler
+                    .findConverterByType(ThreadBorderRouterManagementConverter.class);
+            if (converter != null) {
+                try {
+                    String dataset = Objects.requireNonNull(converter.getActiveDataset().get(),
+                            "Could not get active dataset");
+                    converter.updateThreadConfiguration(dataset);
+                    return new TreeMap<String, Object>() {
+                        {
+                            put("result", ResourceHelper
+                                    .getResourceString(MatterBindingConstants.THING_ACTION_RESULT_SUCCESS));
+                            put("dataset", dataset);
+                        }
+                    };
```

Comment:
This generates a warning (_"The serializable class  does not declare a static final serialVersionUID field of type long"_) because you are creating an anonymous subclass of a serializable class. Would this work instead?
```suggestion
                    return Map.of("result",
                            ResourceHelper.getResourceString(MatterBindingConstants.THING_ACTION_RESULT_SUCCESS),
                            "dataset", dataset);
```